### ISSUE TITLE
Exporter Modifies Date/Time Stamps of Existing Image Files

### ISSUE THREAD

**Issue by Ziflin** [KhronosGroup/glTF-Blender-IO#743]
It appears that (at least on Windows) the glTF exporter is _always_ stomping on the date/time stamp of image files when exporting as "glTF Separate (.glTF + .bin + textures)". This is causing issues with other software that uses the timestamp as an indication that the files have been modified (causing them all to reload, etc.).

**To Reproduce**
1. Apply a simple diffuse texture to the default Cube.
2. Note the "Date modified" time of the texture file.
2. Export as "glTF Separate (.glTF + .bin + textures)"
3. Note the "Date modified" time of the texture file has been modified to match the .glTF and .bin files.

**Expected behavior**
If the texture files already exist on disk, the exporter should _not_ modify them in any way during export.

**Version**
 - OS: [Windows]
 - Blender Version: 2.81 Alpha (2019-10-09 07:42)

**Comment by scurest** [comment]
Exporting to the same directory the image file is located in?

I think the exporter is completely overwriting the file. Apparently when it decides to write an image to the export directory, it will completely clobber any existing file with the same file. That could cause some pretty bad data loss :/

**Comment by Ziflin** [comment]
Exporting to the same directory (so all paths are relative) causes it to stomp on the modified time. It does not appear to actually stomp on the contents of the file if it exists. If I delete the file, the exporter appears to re-save it which is probably acceptable.

So it's a good thing that it doesn't stomp on existing files, but it really should not change the modified times as that causes other applications to think it has been modified on disk...

**Comment by Ziflin** [comment]
@scurest So I looked at this briefly and `io_scene_gltf2/blender/exp/gltf2_blender_gltf2_exporter.py` in `finalize_images(...)` appears to always write out any image files that it gathered. So it appears that it *is* (?) actually loading the images during export only to save them back out to the same file. (My confusion was thinking that it was trying to save the already in-Blender-memory images, but I believe now that it's simply read and immediately writing the same file.)

In order to replace our FBX workflow, we need the exporter to simply ignore any external image file processing. It doesn't need to _read_ or try to convert any data from them. It simply needs to fill in the glTF JSON with their URI. 

Ideally if the export mode is set to "SEPARATE" it would simply skip the redundant reading and writing of image files as I think most (?) people would prefer not to have their files potentially stomped on/reloaded underneath them. This should be a fairly common workflow for most game development.

**Comment by emackey** [comment]
Agreed this exporter should never stomp on pre-existing files.  We do need it to convert files under various circumstances, for example when a user includes a texture that's not JPG or PNG, when a user includes separate metallic and roughness images, separate BaseColor and Alpha images, that sort of thing.  Also, if you're overwriting a glTF with a newer copy of the same glTF, we want it to overwrite the old glTF's converted images with the newly converted images.  It's a challenge to figure out the difference between artist-supplied images and output images from previous runs, but one clue could be to check if the image is used in the blend project itself.

Until this is fixed though, I would strongly recommend that people export to a dedicated export folder, when using the "separate" mode export.

**Comment by Ziflin** [comment]
So I edited this bug's name and description now that I know what's happening. What I think should happen is that there should be a check made in `gltf2_blender_gather_image.py` / `gather_image()` to see `if destFilename == srcFilename` and skip calling `__get_image_data()`. `__gether_uri()` should simply set `data=None` when it creates the `ImageData` instance (as `image_data` would be `None`).

Then in `gltf2_blender_gltf2_exporter.py` the `finalize_images()` method should only attempt to write the image if `image.data != None`.

The filename comparison may need to be normalized/etc. to ensure a correct comparison, but this seems like a fairly easy fix that shouldn't require adding another option to the exporter and it should speed up export times when exporting models with several large PBR textures.

Since I'm not very Python-savvy, the best I was able to do for now was to simply skip reading and writing _all_ images if the mode is set to _glTF Separate_, but some may want to export to a new directory and have the existing functionality still work.

Exporting to a different folder isn't an option for us right now as we need to maintain a single copy of our textures.

**Comment by Ziflin** [comment]
I'm not sure if anyone had additional opinions on how else to fix this issue, but it looks like the `os.path.samefile()` could be used to verify the source and destination files aren't the same and prevent the exporter from reading and then re-writing textures. We'd like to get rid of our current hack/patch at some point though.
